library/ssl_msg.c

```diff
@@ -894,6 +950,34 @@ int mbedtls_ssl_encrypt_buf( mbedtls_ssl_context *ssl,
                             rec->data_len, transform->ivlen,
                             padlen + 1 ) );
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+        status = psa_cipher_encrypt_setup( &cipher_op,
+                                    transform->psa_key_enc, transform->psa_alg );
+
+        if( status != PSA_SUCCESS )
+            return( psa_ssl_status_to_mbedtls( status ) );
+
+        status = psa_cipher_set_iv( &cipher_op, transform->iv_enc, transform->ivlen );
+
+        if( status != PSA_SUCCESS )
+            return( psa_ssl_status_to_mbedtls( status ) );
+
+        status = psa_cipher_update( &cipher_op,
```

Comment:
Suggestion: We're also losing debug prints on any problems that happen here (compared to the non-psa version below). Please consider adding some logs.